cdci_data_analysis/flask_app/dispatcher_query.py

```diff
@@ -1047,18 +1056,21 @@ def update_token(self, update_email_options=False, refresh_token=False):
 
         return self.token
 
-    def download_products(self):
+    def download_file(self):
         try:
             # TODO not entirely sure about these
             self.off_line = False
             self.api = False
 
             self.validate_job_id(request_parameters_from_scratch_dir=True)
             file_list = self.args.get('file_list').split(',')
-            file_name = self.args.get('download_file_name')
+            file_name = self.args.get('download_file_name', None)
+            if file_name is None:
```

Comment:
is this correctly preventing download of anything outside the product directory?

Reply:
This arg was intended to be used as the name of the archive that will contain all the file of the product.

in the case of wanting to download a single file, with the `download_file` endpoint, the name of the downloaded file will match with the desired file that is the arg `file_list`, and it won't create an archive if wished (using the `return_archive` arg)

Reply:
To better answer your question: the path of the file to download is built here:

https://github.com/oda-hub/dispatcher-app/blob/617110283afddc6e862f0156c5a32d9af3cae62b/cdci_data_analysis/flask_app/dispatcher_query.py#L924-L925

So, do we want to check that each single file is present inside the scratch_dir first ?

Reply:
> is built here

Just a side note. There is no need in `+ '/'` here

Reply:
no, you're right. but this was the first time I was looking at that function...I will adjust it

Reply:
> is this correctly preventing download of anything outside the product directory?

No, I just checked. We need to fix it for the dispatcher in general

Reply:
https://github.com/oda-hub/dispatcher-app/pull/667